### DISCUSSION TITLE
Wezterm imgcat command

### DISCUSSION THREAD

**Discussion by idr4n** [wezterm/wezterm#2423] [Ideas]
Would it be possible that the `wezterm imgcat` command allows for `x` and `y` coordinates to position the image in the terminal.

I tried to use Wezterm's image viewer to preview images in LF but I couldn't. However, Kitty's `kitty +icat` command does work for me. I am guessing it is because Kitty's command allows for `--place` x, and y options, in addition to width and height.

Kitty's command works in both Kitty and WezTerm though, but WezTerm protocol implementation seems to be a bit better to render the images.

Thanks in advance!

**Comment by wez** [comment]
At the moment, `wezterm imgcat` uses only the iTerm2 image protocol, which doesn't have any explicit placement control.
You might consider writing a little wrapper script using `tput` to move the cursor position: https://bash.cyberciti.biz/guide/Moving_the_cursor_with_tput

I'm open to the idea of adding flags for this to the imgcat subcommand, and also to giving it the option of using the kitty image protocol instead of the iterm2 protocol.

**Reply by idr4n** [reply]
Thanks for the prompt reply as usual! `tput cup` works very well in the terminal and the following script positions the image as desired:

```sh
#!/usr/bin/env bash

file="$HOME/Wallpapers-Selected/0001.jpg"
x=20
y=2
w=50
h=10

# either of the three below
tput cup 5 "${x}" && wezterm imgcat --width "${w}" --height "${h}" "$file"
tput cup 5 "${x}" && viu -w "${w}" -h "${h}" "$file"
kitty +icat --silent --transfer-mode file --place "${w}x${h}@${x}x5" "$file"
```

Either of the three commands work when I run the script. Unfortunately, that didn't solve the issue to preview images in LF. I can preview images in LF with `kitty +icat --silent --transfer-mode file --place "${w}x${h}@${x}x5" "$file"` but not with `viu` or `wezterm imgcat`. That probably means that the issue is not related to positioning the image inside LF correctly.

Perhaps there is something else going on inside LF when previewing images, and so don't worry, I will not request to add any additional flags to the `wezterm imgcat` command unless is something you might find useful.

**Reply by wez** [reply]
What is `LF` ?

**Reply by idr4n** [reply]
It is a terminal file manager written in Go, similar to Ranger and nnn. https://github.com/gokcehan/lf

**Comment by m-althaf** [comment]
any updates on this @wez

> I'm open to the idea of adding flags for this to the imgcat subcommand, and also to giving it the option of using the kitty image protocol instead of the iterm2 protocol.

this would be great!

Doubts
---
how do we clear image once it's drawn? kitty has clear subcommand for icat

**Reply by wez** [reply]
No one is working on this. No one has filed an enhancement issue for this.

**Reply by Blacksmoke16** [reply]
With https://github.com/wez/wezterm/issues/3716 now released, any luck getting `imgcat` working as your `lf` previewer? 

EDIT: Best I got so far was adding the following to my `lfrc`:
```rc
set previewer pistol
set sixel true
```
Then add the following line to my `pistol.conf`:
```conf
image/* chafa --relative on -f sixel -s %pistol-extra0%x%pistol-extra1% --animate off --polite on %pistol-filename%
```
And this seems to work well enough so far. Would be more ideal if we didn't need the extra lib, but I'll take it for now :+1:.
